File: carbon/src/main/java/carbon/component/BottomSheetCell.java
package carbon.component;

import android.view.ViewGroup;

import carbon.R;
import carbon.beta.BottomSheetLayout;
import carbon.databinding.CarbonBottomsheetCellBinding;

public class BottomSheetCell extends LayoutComponent<BottomSheetLayout.Item> {

    private final CarbonBottomsheetCellBinding binding;

    public BottomSheetCell(ViewGroup parent) {
        super(parent, R.layout.carbon_bottomsheet_cell);
        binding = CarbonBottomsheetCellBinding.bind(getView());
    }

    @Override
    public void bind(BottomSheetLayout.Item data) {
        super.bind(data);
        binding.carbonItemIcon.setImageDrawable(data.getIcon());
        if (data.getIconTintList() != null)
            binding.carbonItemIcon.setTintList(data.getIconTintList());
        binding.carbonItemText.setText(data.getTitle());
    }
}

File: carbon/src/main/java/carbon/component/BottomSheetCell.kt
package carbon.component

import android.view.ViewGroup
import carbon.R
import carbon.beta.BottomSheetLayout
import carbon.databinding.CarbonBottomsheetCellBinding

class BottomSheetCell(parent: ViewGroup) : LayoutComponent<BottomSheetLayout.Item>(parent, R.layout.carbon_bottomsheet_cell) {
    private val binding = CarbonBottomsheetCellBinding.bind(view)

    override fun bind(data: BottomSheetLayout.Item) {
        binding.carbonItemIcon.setImageDrawable(data.icon)
        if (data.iconTintList != null)
            binding.carbonItemIcon.setTintList(data.iconTintList)
        binding.carbonItemText.text = data.title
    }
}